src/health_checker.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+#ifndef HEALTH_CHECKER_H_
+#define HEALTH_CHECKER_H_
+
+#include <boost/algorithm/string/join.hpp>
+#include <string>
+#include <vector>
+#include <map>
+#include <memory>
+#include <fstream>
+
+#include "configuration.h"
+
+namespace {
+  constexpr const char* kHealthStates[] = {
+    "kubernetes_pod_thread", "kubernetes_node_thread"
+  };
+}
+namespace google {
+
+class FileWrapper {
+ public:
+  FileWrapper(const std::string& directory, const std::string& name)
+       : directory_(directory), name_(name) {
+         Touch(directory_, name_);
+  }
+
+  static void Touch(const std::string& directory, const std::string& name) {
+    std::string path(boost::algorithm::join(
+        std::vector<std::string>{directory, name}, "/"));
+    std::ofstream health_file(path);
+    health_file << std::endl;
+  }
+
+  static void Remove(const std::string& directory, const std::string& name) {
+    std::string path(boost::algorithm::join(
+        std::vector<std::string>{directory, name}, "/"));
+    std::remove(path.c_str());
+  }
+
+  static bool Exists(const std::string& directory, const std::string& name) {
```

Comment:
Why do you need this? Why not just use the existence of a `FileWrapper` object in the map instead?